proxy/api/src/http/source.rs

```diff
@@ -729,10 +730,11 @@ mod test {
         })?;
 
         http::test::assert_response(&res, StatusCode::OK, |have| {
-            assert_eq!(have, json!(want));
-            assert_eq!(have.as_array().unwrap().len(), 14);
+            assert_eq!(have["headers"], json!(want.headers));
```

Comment:
I feel like this test should simply be `assert_eq!(have, json!(want))`, is there a reason for the extra cases?

Reply:
You are right! I just extended the tests without re-thinking the existing ones